records/2024-06-04.manifests-kustomize-cicd.md

```diff
@@ -0,0 +1,134 @@
+# Current Manifests CI/CD
+
+Date: 2024-06-04
+
+## Status
+
+**DRAFT**.
+
+## Related ADRs
+
+For information on how manifest rollouts are done using GitHub ARC Runners please see this ADR:
+[Private EKS Cluster](2024-02-15.private-eks-cluster.md)
+
+## Context
+
+The way the CI/CD works for releases for manifests is not well documented. This ADR outlines the the CI/CD release process for staging and production, the current problems we face with this model, and proposed solutions moving forward.
+
+## Current State
+
+### Staging
+
+#### Individual Application Rollouts
+
+When deploying a new version of the application components (admin/api/document-download etc), the release to staging is kicked off at the application repository level when the docker image is built. The docker image tag is based on the first 7 digits of the github runner sha, and also tags this build as latest. The docker build and push workflow invokes a corresponding workflow in the notification-manifests repository passing the new docker tag to have it deployed to staging.
+
+The manifests rollout workflow receives the new image tag and performs a manual patch of the existing deployment:
+
+```shell
+kubectl set image deployment.apps/api api=$DOCKER_SLUG:$DOCKER_TAG -n=notification-canada-ca 
+```
+
+##### Diagram
+
+![Database activity stream architecture](./diagrams/2024-06-04.manifests-current-cicd/staging-cicd.png)
```

Comment:
Could you make it a clearer what is a github action and what isn't in this diagram (maybe even include the workflow filename)?

Also, an idea for any work done on this would be to rename the github workflow files so that at a glance we know what they do, i.e. if its part of a release flow, prefix it with `release_` , if its something to do with testing a PR, prefix it with `testing_`, etc.